File: content-solutions/bfs/frog-leaps.java
import java.util.*;

class Solution {
    public int solution(int[] leap) {
        int n = leap.length;
        int[] dist = new int[n];
        Arrays.fill(dist, -1);
        dist[0] = 0;
        Queue<Integer> queue = new ArrayDeque<>();
        queue.offer(0);
        while (!queue.isEmpty()) {
            int i = queue.poll();
            for (int j : new int[] {i + leap[i], i - leap[i]}) {
                if (j >= 0 && j < n && dist[j] == -1) {
                    dist[j] = dist[i] + 1;
                    queue.offer(j);
                }
            }
        }
        return dist[n - 1];
    }
}

File: content-solutions/bfs/frog-leaps.js
function solution(leap) {
  const n = leap.length;
  const dist = new Array(n).fill(-1);
  dist[0] = 0;
  const queue = [0];
  let head = 0;
  while (head < queue.length) {
    const i = queue[head++];
    for (const j of [i + leap[i], i - leap[i]]) {
      if (j >= 0 && j < n && dist[j] === -1) {
        dist[j] = dist[i] + 1;
        queue.push(j);
      }
    }
  }
  return dist[n - 1];
}
